mongoengine/queryset/manager.py

```diff
@@ -24,6 +27,7 @@ def __init__(self, queryset_func=None):
         if queryset_func:
             self.get_queryset = queryset_func
 
+    @retry
```

Comment:
You retry the whole instantiation of a new QuerySet. That seems a bit excessive. Perhaps there's a better place where we could add the retrying mechanism? Also, what happens if the primary disconnects when for example we call `qs.first()` on a query set that's been instantiated before the connection issues started? Will that work?